server/repository/src/migrations/v2_04_00/add_demographic_table.rs

```diff
@@ -0,0 +1,46 @@
+use crate::migrations::*;
+
+pub(crate) struct Migrate;
+
+impl MigrationFragment for Migrate {
+    fn identifier(&self) -> &'static str {
+        "add_demographic_table"
+    }
+
+    fn migrate(&self, connection: &StorageConnection) -> anyhow::Result<()> {
+        // NOTE: These migrations assume that we've only ever created demographic_projection/indicators for a single base year.
+        // This should be safe as we haven't had a live release yet.
+
+        // Create demographic table
+        sql!(
+            connection,
+            r#"
+                CREATE TABLE demographic (
+                    id TEXT NOT NULL PRIMARY KEY,
+                    name TEXT NOT NULL
+                );
+            "#
+        )?;
+
+        // Insert any existing data into the new table
+        // We'll assume we don't have duplicate names (yet!)
```

Comment:
In the very low likelihood that somewhere, someone plays with demographics before upgrading to 2.4, and might end up with duplicates here... there's no major impact right? There would be duplicate named records, but we could update the indicators to point to one, then delete the other(s)? Or am I missing other implications 👀

Reply:
Yeah I don't think it's too bad, problem if it did happen is we don't have soft delete, or sync for deletes so we can't easily clean up remote sites. I think it's pretty unlikely particularly if we get it in `2.3.1`